staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -113,6 +113,7 @@ func (e *eventBroadcasterImpl) refreshExistingEventSeries() {
 	defer e.mu.Unlock()
 	for isomorphicKey, event := range e.eventCache {
 		if event.Series != nil {
+			event.DeprecatedLastTimestamp = metav1.Time{Time: event.Series.LastObservedTime.Time}
```

Comment:
Hmm - those are supposed to be handled here:
https://github.com/kubernetes/kubernetes/blob/master/pkg/printers/internalversion/printers.go#L1719

I think we should rather fix the printer itself to fix this issue if that logic isn't correct.

Reply:
`event.Series` is nil the first time the event is created. After that it takes minutes (given by `refreshTime`, resp. `finishTime`) before the event is updated (with `event.Series` getting set). Until that, `event.Series` is still nil and there's no way to deduce `LastTimestamp`.

Reply:
I don't understand. Series field will be updated on each subsequent update of the event.

Reply:
One of the important decision behind the API was exactly to not send every single event. So yes, the LastUpdate often will not be accurate to the time when it really happened in practice, but those events won't be send to apiserver thus there is no way to detect this time.